plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeDynamicFiltering.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.trino.Session;
+import io.trino.execution.DynamicFilterConfig;
+import io.trino.execution.QueryStats;
+import io.trino.operator.OperatorStats;
+import io.trino.plugin.deltalake.util.DockerizedMinioDataLake;
+import io.trino.spi.QueryId;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import static com.google.common.base.Verify.verify;
+import static io.airlift.testing.Assertions.assertGreaterThan;
+import static io.trino.SystemSessionProperties.ENABLE_DYNAMIC_FILTERING;
+import static io.trino.plugin.deltalake.DeltaLakeDockerizedMinioDataLake.createDockerizedMinioDataLakeForDeltaLake;
+import static io.trino.plugin.deltalake.DeltaLakeQueryRunner.DELTA_CATALOG;
+import static io.trino.sql.planner.OptimizerConfig.JoinDistributionType.PARTITIONED;
+import static io.trino.tpch.TpchTable.LINE_ITEM;
+import static io.trino.tpch.TpchTable.ORDERS;
+import static java.lang.String.format;
+
+public class TestDeltaLakeDynamicFiltering
+        extends AbstractTestQueryFramework
+{
+    private static final String BUCKET_NAME = "delta-lake-test-dynamic-filtering";
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        verify(new DynamicFilterConfig().isEnableDynamicFiltering(), "this class assumes dynamic filtering is enabled by default");
+        DockerizedMinioDataLake dockerizedMinioDataLake = closeAfterClass(createDockerizedMinioDataLakeForDeltaLake(BUCKET_NAME));
+        QueryRunner queryRunner = DeltaLakeQueryRunner.createS3DeltaLakeQueryRunner(
+                DELTA_CATALOG,
+                "default",
+                // Slowing down the query ensures the dynamic filter has enough time to populate.
+                ImmutableMap.of("delta.max-splits-per-second", "3"),
```

Comment:
We end up needing this and the large table on build side as we don't have a way to make delta lake split manager wait on DF (similar to dynamic-filtering.wait-timeout in hive and iceberg). It would be great to have that as it would simplify the testing and make it less prone to being flaky.

Reply:
Compared to Iceberg, `DeltaLakeSplitSource` produces the splits asynchronously. Does the `dynamicFilteringWaitTimeoutMillis` concept still apply in this case?

Reply:
Yes, HiveSplitSource produces splits asynchronously as well.

Reply:
Created https://github.com/trinodb/trino/issues/11600 follow up issue